src/Tools/IiifHelper.php

```diff
@@ -13,46 +13,50 @@
 class IiifHelper {
 
   use StringTranslationTrait;
-  
-  protected $remoteInfoJsonURL;
 
+  protected $publicUrl;
+  protected $identifier;
+  protected $internalUrl;
 
   /**
    * @var array;
    */
-  protected $remoteInfoJson;
+  protected $remoteInfoJsonData;
 
   /**
    * @var \GuzzleHttp\Client;
    */
   protected $httpClient;
 
 
-
-  public function __construct($remoteUrl) {
+  public function __construct($publicUrl, $internalUrl, $identifier) {
     //@TODO verify that remoteURL is a valid, actual IIIF info json URL.
-    $this->remoteInfoJsonURL = $remoteUrl;
     $this->httpClient = \Drupal::httpClient();
-    $this->remoteInfoJson = $this->getRemoteIIIFImageInfo($remoteUrl);
+    $this->remoteInfoJsonData = $this->getRemoteInfoJsonData();
+    $this->publicUrl = $publicUrl;
+    $this->internalUrl = $internalUrl;
+    $this->identifier = $identifier;
   }
 
-  public function getRemoteIIIFImageInfo($remoteUrl) {
```

Comment:
Nope. Don't like the method renaming here. Its a public method so already exposed. Also ImageInfo is accurate. You will have to make a good point here to have that one changed!!! 😄

Reply:
I understand about not renaming public functions! So, ok. Reading the code I was getting confused between "URL" "Json" "InfoJson" "Info" -- I was trying to make it more consistent. Ok I learned, an InfoJson is a URL that has json data. Sometimes you need the URL, sometimes you  need the data you get from the URL. ETc. getRemoteIIIFImageInfo just confused me, it was unrelated to the other naming!